File: 0009-palindrome-number/simple.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if (x < 0):
            return False
        else:
            original = x
            reversed = 0

            LIMIT = 214748364

            # Store digits in reverse order in "reversed"
            while (x != 0):
                digit = x % 10

                # Check if we're going to overflow
                if ((reversed > LIMIT) or ((reversed == LIMIT) and (digit > 7))):
                    return False
                else:
                    reversed = (10 * reversed) + digit
                    x = x // 10
            return (original == reversed)

File: 0009-palindrome-number/simple.rs
impl Solution
{
    pub fn is_palindrome(x: i32) -> bool
    {
        if (x < 0)
        {
            return false;
        }
        else
        {
            let original: i32 = x;
            let mut n: i32 = x;
            let mut reversed: i32 = 0;

            const LIMIT: i32 = 214748364;

            // Store digits in reverse order in "reversed"
            while (n != 0)
            {
                let digit: i32 = n % 10;

                // Check for overflow
                if ((reversed > LIMIT) || ((reversed == LIMIT) && (digit > 7)))
                {
                    return false;
                }
                else
                {
                    reversed = (10 * reversed) + digit;
                    n = n / 10;
                }
            }

            return (original == reversed);
        }
    }
}